lib/stm32/l4/rcc.c

```diff
@@ -479,7 +482,20 @@ uint32_t rcc_get_usart_clk_freq(uint32_t usart)
 uint32_t rcc_get_timer_clk_freq(uint32_t timer)
 {
 	/* Handle APB1 timers, and apply multiplier if necessary. */
-	if (timer >= TIM2_BASE && timer <= TIM7_BASE) {
+	if (timer == LPTIM1_BASE || timer == LPTIM2_BASE) {
+		int shift = (timer == LPTIM1_BASE) ? RCC_CCIPR_LPTIM1SEL_SHIFT : RCC_CCIPR_LPTIM2SEL_SHIFT;
+		uint8_t clksel = (RCC_CCIPR >> shift) & RCC_CCIPR_LPTIMxSEL_MASK;
+		switch (clksel) {
+			case RCC_CCIPR_LPTIMxSEL_APB:
+				return rcc_apb1_frequency;
+			case RCC_CCIPR_LPTIMxSEL_LSI:
+				return 32000;
+			case RCC_CCIPR_LPTIMxSEL_HSI16:
+				return 16000000U;
+			case RCC_CCIPR_LPTIMxSEL_LSE:
+				return 32768;
```

Comment:
why have you mixed explicitly unsigned with signed here?

Reply:
It's copy and paste and I think I read it as UL because that would somewhat make sense. Just U does not really help here AFAICT. With UL it makes clear that even on 16-bit machines the value should not be cut (not sure what the rules would be when the literal is directly used in a return statement of type `uint32_t` though), something that cannot happen with 32768 at all. However, that's not really relevant to this library anyway...
Counter question: why are unsigned literals used in the library? :)
Do you prefer them all to be unsigned?

Reply:
please do remember the target here.  this is a cortex-m library, there is _no concept_ of int being 16bit here, and we're not going "just be safe" for that sort of thing :)

They're unsigned in many places because we are often shifting things, and shifting ints has some UB that was triggering warnings.

Reply:
I do remember, that's why I said it's not really relevant. :) There won't be a lot of shifting when returning a literal though, so I am a bit puzzled why the suffix is used so often there. I'll add it to make it homogeneous.

Reply:
Personally, I don't add them unless I get warnings, all I was commenting on here was the inconsistency :)